src/main/java/uk/gov/hmcts/divorce/citizen/validation/rules/caseData/MarriageDate.java

```diff
@@ -0,0 +1,84 @@
+package uk.gov.hmcts.divorce.citizen.validation.rules.caseData;
+
+import com.deliveredtechnologies.rulebook.annotation.Given;
+import com.deliveredtechnologies.rulebook.annotation.Result;
+import com.deliveredtechnologies.rulebook.annotation.Then;
+import com.deliveredtechnologies.rulebook.annotation.When;
+
+import lombok.Data;
+import uk.gov.hmcts.divorce.common.model.CaseData;
+import uk.gov.hmcts.divorce.common.utils.DateUtils;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Data
+public class MarriageDate {
+
+    private static final String BLANK_SPACE = " ";
+    private static final String ACTUAL_DATA = "Actual data is: %s";
+    private static final String ERROR_MESSAGE_NULL = "MarriageDate can not be null or empty.";
+    private static final String ERROR_MESSAGE_LESS_THAN_ONE_YEAR_AGO =
+        "MarriageDate can not be less than one year ago.";
+    private static final String ERROR_MESSAGE_MORE_THAN_ONE_HUNDRED_YEARS_AGO =
+        "MarriageDate can not be more than 100 years ago.";
+    private static final String ERROR_MESSAGE_IN_THE_FUTURE = "D8MarriageDate can not be in the future.";
+
+    @Result
+    public List<String> result;
+
+    @Given("caseData")
+    public CaseData caseData = new CaseData();
+
+    @When
+    public boolean when() {
+        String marriageDate = caseData.getMarriageDate();
```

Comment:
If the marriageDate field is already converted to LocalDate by the deserializer, it would probably make the validations easier.  See createdDate for an example of the Json annotations needed.